src/test/java/jenkins/bouncycastle/EncodignDecodingTest.java

```diff
@@ -224,5 +226,15 @@ public void testCreationFromObjectPrivateKeyPEM() throws Exception {
 
         assertEquals(FileUtils.readFileToString(PRIVATE_KEY_PEM), FileUtils.readFileToString(pemFileNew));
     }
+    
+    
+    @Test
+    @Issue(value="JENKINS-35661") 
+    public void testReadKeyPairFromPCKS8PEM() throws Exception {
+        PEMEncodable pemEnc = PEMEncodable.read(getResourceFile("private-key-pcks8.pem"));
+        assertNotNull(pemEnc.toKeyPair());
+        assertNotNull(pemEnc.toPrivateKey());
+        assertNotNull(pemEnc.toPublicKey());
```

Comment:
@stephenc  I've created an additional test with a key that has public exponen 3 (the allowed alternative with `openssl genrsa`), it is working correctly.